user/user.go

```diff
@@ -83,3 +78,60 @@ func userCreateHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	json.NewEncoder(w).Encode(response)
 }
+
+func userUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	userID, err := ExtractUserIDFromRequest(mux.Vars(r))
+	if err != nil {
+		http.Error(w, CreateErrorJSON(err.Error()), http.StatusBadRequest)
+		return
+	}
+
+	var req dao.UserUpdateRequest
+	err = json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		errMsg := CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	_, err = valid.ValidateStruct(req)
+	if err != nil {
+		errMsg := CreateErrorJSON(fmt.Sprintf("Invalid request parameters: %s", err.Error()))
```

Comment:
Can this whole block be abstracted out? Seems like we do it a lot - maybe for another PR though

Reply:
Is this happening this PR or next?

Reply:
Probably the next